odoo/addons/base/models/assetsbundle.py

```diff
@@ -328,11 +330,23 @@ def js(self):
         if not js_attachment:
             template_bundle = ''
             if self.templates:
-                content = ['<?xml version="1.0" encoding="UTF-8"?>']
-                content.append('<templates xml:space="preserve">')
-                content.append(self.xml(show_inherit_info=not is_minified))
-                content.append('</templates>')
-                templates = '\n'.join(content).replace("\\", "\\\\").replace("`", "\\`").replace("${", "\\${")
+                content = []
+                templates_dict = {}
+                try:
+                    templates_dict = self.xml(show_inherit_info=not is_minified)
+                except XMLAssetError as e:
+                    content.append(f'throw new Error({json.dumps(to_text(e))});')
+
+                for _, templates in templates_dict.items():
+                    for name, (element, url) in templates.items():
+                        if 't-name' in element.attrib:
+                            name = element.attrib['t-name']
+                            string = etree.tostring(element, encoding='unicode')
+                            if not is_minified:
+                                content.append("/* Filepath: %s */" % ' => '.join(url))
+                            # Add a '#' before the backtick to avoid the backtick to be replaced by \\backtick
+                            content.append(f'registerTemplate("{name}", #`{string}#`);')
+                templates = '\n'.join(content).replace("\\", "\\\\").replace("`", "\\`").replace("${", "\\${").replace("#\\`", "`")
```

Comment:
Seems like a weird hack, shouldn't we just escape the name and template separately? You can probably just `json.dumps` both of them and omit the outer quotes (unless we care about preserving the newlines inside the templates as newlines instead of escape sequences)